drivers/cc2420/cc2420_getset.c

```diff
@@ -0,0 +1,248 @@
+    if (state) {
+        dev->options |= option;
+        /* trigger option specific actions */
+        switch (option) {
+            case CC2420_OPT_CSMA:
+                DEBUG("[cc2420] opt: enabling CSMA mode (NOT IMPLEMENTED)\n");
+                /* TODO: en/disable csma */
+                break;
+            case CC2420_OPT_PROMISCUOUS:
+                DEBUG("[cc2420] opt: enabling PROMISCUOUS mode\n");
+                tmp = cc2420_reg_read(dev, CC2420_REG_MDMCTRL0);
+                /* disable auto ACKs in promiscuous mode */
+                tmp &= ~CC2420_AUTOACK;
+                /* enable promiscuous mode */
+                tmp &= ~CC2420_ADR_DECODE;
+                cc2420_reg_write(dev, CC2420_REG_MDMCTRL0, tmp);
+                dev->options &= ~(CC2420_OPT_AUTOACK);
+                break;
+            case CC2420_OPT_AUTOACK:
+                DEBUG("[cc2420] opt: enabling auto ACKs\n");
+                tmp = cc2420_reg_read(dev, CC2420_REG_MDMCTRL0);
+                tmp |= CC2420_AUTOACK;
+                cc2420_reg_write(dev, CC2420_REG_MDMCTRL0, tmp);
+                break;
+            case CC2420_OPT_TELL_RX_START:
+                DEBUG("[cc2420] opt: enabling SFD IRQ (NOT IMPLEMENTED)\n");
+                /* TODO */
+                break;
+            default:
+                /* do nothing */
+                break;
+        }
+    }
+    else {
+        dev->options &= ~(option);
+        /* trigger option specific actions */
+        switch (option) {
+            case CC2420_OPT_CSMA:
+                DEBUG("[cc2420] opt: disabling CSMA mode (NOT IMPLEMENTED)\n");
+                /* TODO: en/disable csma */
+                break;
+            case CC2420_OPT_PROMISCUOUS:
+                DEBUG("[cc2420] opt: disabling PROMISCUOUS mode\n");
+                /* disable promiscuous mode */
+                tmp = cc2420_reg_read(dev, CC2420_REG_MDMCTRL0);
+                tmp |= CC2420_ADR_DECODE;
+                /* re-enable AUTOACK only if the option was set */
+                if (dev->options & CC2420_OPT_AUTOACK) {
+                    tmp |= CC2420_AUTOACK;
+                }
+                cc2420_reg_write(dev, CC2420_REG_MDMCTRL0, tmp);
+                break;
+            case CC2420_OPT_AUTOACK:
+                DEBUG("[cc2420] opt: disabling auto ACKs\n");
+                tmp = cc2420_reg_read(dev, CC2420_REG_MDMCTRL0);
+                tmp &= ~CC2420_AUTOACK;
+                cc2420_reg_write(dev, CC2420_REG_MDMCTRL0, tmp);
+                break;
+            case CC2420_OPT_TELL_RX_START:
+                DEBUG("[cc2420] opt: disabling SFD IRQ (NOT IMPLEMENTED)\n");
+                /* TODO */
+                break;
+            default:
+                /* do nothing */
+                break;
+        }
+    }
+}
+
+inline uint8_t cc2420_get_state(cc2420_t *dev)
+{
+    uint16_t state = 0;
+    state = cc2420_reg_read(dev, CC2420_REG_FSMSTATE);
+    return dev->state;
```

Comment:
Shouldn't you return the `state` variable, instead of the `dev->state` field? Or do something like:
```
    dev->state = state;
    return dev->state;
```
It seems it would make much more sense.
Plus, I think this one of the problems that make Travis static tests fail in the first place...